cli/dcoscli/marathon/main.py

```diff
@@ -674,15 +678,20 @@ def _clean_up_resource_definition(properties):
     :rtype: dict
     """
     clean_properties = {}
+    ignore_properties = ["deployments", "lastTaskFailure", "tasks",
+                         "tasksRunning", "tasksHealthy", "tasksUnhealthy",
+                         "tasksStaged"]
     for k, v in properties.items():
-        if v:
-            if k in ["apps", "groups"]:
-                clean_properties[k] = [_clean_up_resource_definition(v[0])]
-            # Remove fields that marathon adds. These aren't in the json-schema
-            # because a user should not specify them. When we do a `get` to see
-            # the current schema, we may see these fields, but leaving them
-            # will cause us to unnecessarily fail the schema validation.
-            elif not k.startswith("task") and k != "lastTaskFailure":
+        if v is not None and k not in ignore_properties:
+            # iterate through dict/list(s) to check all properties
+            if isinstance(v, list):
+                clean_properties[k] = [
+                    _clean_up_resource_definition(elem)
+                    if isinstance(elem, dict)
+                    else elem for elem in v if elem is not None]
```

Comment:
This was really hard to read for me. Probably because my Python sucks. Can we improve this?